### DISCUSSION TITLE
Welcome to ic1101 Discussions!

### DISCUSSION THREAD

**Discussion by librick** [librick/ic1101#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bitglitcher** [comment]
Awesome repo. Would be nice to know how the Head Unit communicates with the Dash to display information.

**Reply by bitglitcher** [reply]
I looked at the Documents in the repo and it seems that there is an LVDS Link to the Main Display and the Gauge Cluster. It would be interesting to see if there is any kind of framebuffer you can manually write to in order to display things in the Gauge Cluster.

I'm trying to modify mine in order to include a new Instrument into my instrument cluster. Something like Sensor readings.

**Reply by librick** [reply]
Thank you for the kind words :)

I haven't looked much more into LVDS but I encourage anyone who wants to take a look. I extracted the boot files and the init.rc file contains these conveniently-commented lines on LVDS initialization:
```
# GPIO Setting for LVDS Serializer
    write /sys/class/gpio/export 129
    write /sys/class/gpio/export 130
    write /sys/class/gpio/export 131
    write /sys/class/gpio/export 134
    write /sys/class/gpio/export 135
    write /sys/class/gpio/export 156
    write /sys/class/gpio/export 159
    write /sys/class/gpio/export 176
    write /sys/class/gpio/export 177
    write /sys/class/gpio/gpio156/direction "in"
    write /sys/class/gpio/gpio159/direction "out"
    write /sys/class/gpio/gpio176/direction "out"
    write /sys/class/gpio/gpio177/direction "in"
    write /sys/class/gpio/gpio156/edge "falling"
    write /sys/class/gpio/gpio177/edge "falling"
```

I've had a bit more time recently to dive back into this repo. I've mostly been interested in reverse engineering the HondaLink (formerly known as AcuraLink) protocol for things like remote start, so I don't anticipate working on LVDS documentation specifically, but I wish you the best of luck if you feel like poking around!

**Reply by librick** [reply]
Also, I don't want to link it directly for legal/copyright reasons, but if you Google "civicx MRC_EU_SW_v12_4.zip" you can find a copy of the firmware to tinker on including all of the native Android libraries and APKs.
